bpx/function.py

```diff
@@ -87,26 +86,10 @@ def to_python_function(self, preamble: str | None = None) -> Callable:
         source_code = preamble + function_def + function_body
 
         with tempfile.NamedTemporaryFile(suffix=f"{function_name}.py", delete=False) as tmp:
-            # write to a tempory file so we can
-            # get the source later on using inspect.getsource
-            # (as long as the file still exists)
-            tmp.write((source_code).encode())
+            tmp.write(source_code.encode())
             tmp.flush()
-
-            # Now load that file as a module
             spec = util.spec_from_file_location("tmp", tmp.name)
             module = util.module_from_spec(spec)
             spec.loader.exec_module(module)
 
-            # Delete
-            tmp.close()
-            Path(tmp.name).unlink(missing_ok=True)
-            if module.__cached__:
-                cached_file = Path(module.__cached__)
-                cached_path = cached_file.parent
-                cached_file.unlink(missing_ok=True)
-                if not any(cached_path.iterdir()):
-                    cached_path.rmdir()
-
-        # return the new function object
```

Comment:
This was causing failures due to cached files not being found. It did not seem necessary